### PR TITLE
Assignment7

### PR DESCRIPTION
**Pull request by SharanyaSD**
1. Given a string containing conversation between alice and bob. In the string, if it reaches $, it means end of alice message and if it reaches #, it means end of bob's message. and if it reaches ^,
it means end of conversation ignore string after that.

Note: given string doesn't contain any spaces. If message contains two continuous conversations from one person it should be printed one after another as given in the example. 

write a program to separate out messages from alice and bob. Write messages from alice and bob on seperate channels, whenever a message from alice/bob, print it in front of their name as shown in the example below:

Note: there is a single space before and after colon(:) and no space before and after comma.

e.g: "helloBob$helloalice#howareyou?#Iamgood.howareyou?$^"
output: alice : helloBob,bob : helloalice,bob : howareyou?,alice : Iamgood.howareyou?

2. Given a string, reverse it using one go routine.And inside go routine print reversed string and number of goroutines launched

e.g: Input: test123 output: 321tset 2

### CODE REVIEW

File: A7question2.go
Comment (yama11299) [inline A7question2.go:1]: nice code. very optimized, but there is one bug in this. try input -> test1234世界

Comment (SharanyaSD) [reply A7question2.go:1]: go run A7question2.go
Enter a string: 
test1234%$
Reversed string:  $%4321tset
No. of Go routines launched:  2

Comment (yama11299) [reply A7question2.go:1]: No, did you try unicode characters? no issues leave it for now.

File: A7question1.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	bobChannel:=make(chan string)
+	channelSignal:=make(chan struct{})
+	
+	go func() {
+		string1:="helloBob$helloalice#howareyou?#Iamgood.howareyou?$^"
+
+		string2 :=""
+		for _,symbol :=range string1 {
+
+			if symbol=='$' {
```
Comment (yama11299) [inline A7question1.go:18]: can update it with switch, looks better

**Pull request merged**